### PR TITLE
Feature body force

### PR DESCRIPTION
**Pull request by economon**
Adds a general body force source term implementation to the Finite Volume flow solver, which is particularly useful for pressure-driven / gravity / channel problems, for instance. 

Also includes a simple 2D Poiseuille flow case with an analytic solution for testing the body force and periodic implementations.

### CODE REVIEW

File: Common/include/config_structure.hpp [hunk 1]
```diff
@@ -823,8 +823,11 @@ class CConfig {
   *default_grid_fix,          /*!< \brief Default fixed grid (non-deforming region) array for the COption class. */
   *default_htp_axis,          /*!< \brief Default HTP axis for the COption class. */
   *default_ffd_axis,          /*!< \brief Default FFD axis for the COption class. */
-  *default_inc_crit;          /*!< \brief Default incremental criteria array for the COption class. */
-  
+  *default_inc_crit,          /*!< \brief Default incremental criteria array for the COption class. */
+  *default_body_force;        /*!< \brief Default body force vector for the COption class. */
+  bool Body_Force;  			       /*!< \brief Flag to know if a body force is incuded in the formulation. */
```
Comment (arubino) [inline Common/include/config_structure.hpp]: just a small typo here: ~~incuded~~ -> included

**Comment by economon** [comment]
Thanks, Antonio, will fix. I'll wait for a second reviewer on this one.

File: SU2_CFD/src/numerics_direct_mean.cpp [hunk 2]
```diff
@@ -3993,6 +3993,51 @@ void CSourceGravity::ComputeResidual(su2double *val_residual, CConfig *config) {
...
+
+    /*--- Momentum contribution ---*/
+
+    for (iDim = 0; iDim < nDim; iDim++)
+      val_residual[iDim+1] = -Volume * U_i[0] * Body_Force_Vector[iDim] / Force_Ref;
+
+    /*--- Energy contribution ---*/
+
+    for (iDim = 0; iDim < nDim; iDim++)
+      val_residual[nDim+1] = -Volume * U_i[iDim+1] * Body_Force_Vector[iDim] / Force_Ref;
```
Comment (talbring) [inline SU2_CFD/src/numerics_direct_mean.cpp]: Something is probably not correct here. val_residal[nDim+1] is overwritten in each loop cycle.

Comment (economon) [inline SU2_CFD/src/numerics_direct_mean.cpp]: Code reviews prove their worth again... yep, bug there. Should be accumulating. Fixing now, let's see how it affects the test.

**Pull request merged**